### PR TITLE
Added Transpose, SplitND and ConcatND to model_spec

### PR DESCRIPTION
**Pull request by abhishekpratapa**
Added Layers to the model_spec
- Transpose
- SplitND
- Concat

### CODE REVIEW

File: src/ml/neural_net/model_spec.cpp [hunk 1]
```diff
@@ -975,6 +978,48 @@ void model_spec::add_preprocessing(const std::string& feature_name,
   image_scaler->set_channelscale(image_scale);
 }
 
+void model_spec::add_transpose(const std::string& name,
+                               const std::string& input,
+                               std::vector<size_t> axes) {
```
Comment (guihao-liang) [inline src/ml/neural_net/model_spec.cpp]: just curious why not const vec<>&?

Comment (nickjong) [reply src/ml/neural_net/model_spec.cpp]: Yes, let's make this a const reference

File: src/ml/neural_net/model_spec.cpp [hunk 2]
```diff
@@ -983,8 +1162,7 @@ pipeline_spec& pipeline_spec::operator=(pipeline_spec&&) = default;
 pipeline_spec::~pipeline_spec() = default;
 
 std::unique_ptr<Pipeline> pipeline_spec::move_coreml_spec() && {
-  return std::move(impl_);
-}
+  return std::move(impl_);}
```
Comment (nickjong) [inline src/ml/neural_net/model_spec.cpp]: This introduces a formatting error, I believe. Please run clang-format

**Comment by nickjong** [comment]
Superseded by #3111

**Pull request closed without merging**